spec/setup/helpers/abs_helper_spec.rb

```diff
@@ -1099,5 +1114,105 @@ class AbsHelperClass
       end
     end
   end
+
+  describe "#parse_abs_resource_hosts" do
+    context "when a valid hosts JSON string is specified" do
+      it "returns the hosts array" do
```

Comment:
Parsing a valid JSON string does not necessarily return an array. The method states that is takes a JSON string, but doesn't define what a "valid hosts JSON string" is.
```
irb(main):002:0> require 'json'
=> true
irb(main):003:0> hosts = JSON.parse("{}")
=> {}
```

Reply:
Updated to move the actual validation into this method (raises an error if invalid) and included an example of a valid hosts JSON string in the description.